src/annotator/integrations/html.ts

```diff
@@ -36,8 +62,10 @@ export class HTMLIntegration extends TinyEmitter implements Integration {
   private _htmlMeta: HTMLMetadata;
   private _prevURI: string;
 
-  /** Whether to attempt to resize the document to fit alongside sidebar. */
-  private _sideBySideEnabled: boolean;
+  /** Options to attempt to resize the document to fit alongside sidebar. */
+  private _sideBySideOptions: SideBySideOptions & {
+    initiallyProvidedMode?: SideBySideMode;
+  };
```

Comment:
I think the logic around "merging" the side-by-side configuration with the feature flags can be simplified by storing only the minimal state in the class, and then computing the effective state on demand. In this case the minimal state you'd need is:

- The last-observed value of the feature flag (to know if it changed when `flagsChanged` is emitted)
- The side-by-side mode from the configuration (defaults to `auto` if omitted)

Then in `fitSideBySide` you can compute the effective state as `const active = this._sideBySideFlagEnabled && this._sideBySideMode === 'auto' && haveEnoughRoom`.

When this feature flag changes, you just check whether the side-by-side flag changed, and re-run `fitSideBySide` if it changed. If the mode in the configuration is manual, or there is not enough room, then this will be a no-op.

Reply:
Yep, I think this makes sense. Good suggestion.